sklearn/impute/_iterative.py

```diff
@@ -762,6 +778,13 @@ def fit_transform(self, X, y=None, **params):
         self._min_value = self._validate_limit(self.min_value, "min", X.shape[1])
         self._max_value = self._validate_limit(self.max_value, "max", X.shape[1])
 
+        # Make sure to remove the empty feature elements from the bounds
+        nonempty_feature_mask = np.logical_not(np.all(complete_mask, axis=0))
+        if len(self._min_value) == len(nonempty_feature_mask):
+            self._min_value = self._min_value[nonempty_feature_mask]
+        if len(self._max_value) == len(nonempty_feature_mask):
+            self._max_value = self._max_value[nonempty_feature_mask]
+
```

Comment:
The same here, I think that we should move this code in the `_validate_limit` function.